voting/v2/contracts/Voting.sol

```diff
@@ -376,6 +377,54 @@ contract Voting is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeable
 		return totalWeightByUser;
 	}
 
+	// Get weight for voting by lockFarm, time
+	function getWeightByTime(LockFarm _lockFarm, uint256 _startTime) public view returns (uint256) {
+		uint256 weightByTime = 0;
+		// Calculate user's voting weights in customized time
+		uint256 i = totalFarmVoteCount - 1;
+		do {
```

Comment:
@xiangdev1016 I am curious your thought process behind using the do-while loop while you can do make things cleaner with a simple for loop.

Reply:
what caused the revert? and the solution to fix it?